Formula/mongo-c-driver.rb

```diff
@@ -21,8 +21,11 @@ class MongoCDriver < Formula
 
   def install
     cmake_args = std_cmake_args
-    cmake_args << "-DBUILD_VERSION=1.17.0-pre" if build.head?
+    cmake_args << "-DBUILD_VERSION=1.18.0-pre" if build.head?
     system "cmake", ".", *cmake_args
+    # Replace full stored compiler path.
+    cc_path = `type -path $CC`
+    inreplace "src/libmongoc/src/mongoc/mongoc-config.h", cc_path.chomp, ENV.cc
```

Comment:
libmongoc stores the compiler in the generated mongoc-config.h. This replaces the full path with just the compiler name.

Similar to the issue described here: https://discourse.brew.sh/t/updating-root-formula-a-few-questions/8099